docs/DeveloperGuide.md

```diff
@@ -492,3 +501,68 @@ testers are expected to do more *exploratory* testing.
     1. _{ tbc }_
 
 1. _{ more test cases …​ }_
+
+## **Appendix: Planned Enhancements**
+
+Team size: 5
+
+1. **Enable cross day free time interval:**
+   * Currently, a free time interval can only be specified based on a single day. There is no way to specify a free time interval that spans multiple day.
+   * To improve user experience, it could be useful to enable cross day free time interval, such as `Mon:2000-Tue:0800`.
+
+2. **Enable free time interval merging:**
+   * Currently, a person can have overlapping free time intervals.
+     * For example, a person can have free time intervals of `Mon:1000-1200` and `Mon:1100-1300`, or `Mon:2000-2359` and `Tue:0000-0800`.
+   * To improve user experience, it could be useful to merge free time tags that have overlapping time intervals.
+     * For example, `Mon:1000-1200` and `Mon:1100-1300` can be combined as `Mon:1000-1300`.
+     * For example, `Mon:2000-2359` and `Tue:0000-0800` can be combined as `Mon:2000-Tue:0800`.
+   * The merging process could be done when new free time tag is added to the person, e.g. via `add` command, `addTime` command, or `edit` command.
+
+3. **Delete free time within an interval:**
+   * Additionally, it could be useful to be able to delete free times that fall within a time interval, e.g. via `deleteTime` command.
+   * For example, an initial freeTimeTag could be `Mon:1000-1400`. Executing `deleteTime INDEX ft/Mon:1100-1200` will result in the new freeTimeTags `Mon:1000-1100` and `Mon:1200-1400`.
+
+4. **Allow more flexibility in command prefixes:**
+   * Currently, the command prefixes are case-sensitive.
+     * For example, `n/` is the prefix for "name" parameter, but `N/` is not acceptable by the command parser.
+   * It may provide more convenience to user by disabling case sensitivity for prefixes. For example, allowing `n/` and `N/` as the prefix for "name" parameter.
+
+5. **Supporting more forms of name:**
+   * Currently, only name form that consist of alphanumeric characters and spaces in between is allowed.
+   * It may improve the diversity of the application, by supporting name in different forms, including names in different languages and names that contain non-alphanumeric symbols.
+
+6. **Supporting more forms of room number:**
+   * Currently, the room number format is {block}-{floor}-{room number}, where block and room number are at least 2 alphanumeric characters and floor is strictly 2 alphanumeric characters. E.g. `nw-12-12`.
+   * To accommodate more room number format, it will be good to change the minimum length of block and room number as 1.
+
+7. **Allow more flexibility in free time tag format:**
+   * Currently, free time tag only accept forms of `DDD:HHmm-HHmm`, e.g. `Mon:1200-1300`.
+   * It may provide more convenience to user by disabling case sensitivity on the day part, e.g. allowing `mon:1000-1200`, `Mon:1000-1200`, `MON:1000-1200`.
+
+8. **Support more accurate free time:**
```

Comment:
We might need to only choose one of the solutions, so perhaps we could go with the date one but we can say that its like an extension. So we still have our original free time tags that is recurring but we can also specify a data on top of that if we need. TLDR default free time tag is recurring weekly unless we specify a date

Reply:
Will change in the next update.